packages/edit-site/src/components/style-book/index.js

```diff
@@ -210,8 +246,17 @@ const StyleBookBody = ( {
 		readonly: true,
 	};
 
+	const handleLoad = () => setHasIframeLoaded( true );
+	useLayoutEffect( () => {
+		if ( goTo?.block && hasIframeLoaded && iframeRef?.current ) {
+			scrollToSection( `example-${ goTo?.block }`, iframeRef?.current );
```

Comment:
On page refreshes this might not be 100% accurate. I assume because some of the iframe assets are still being rendered and therefore the document height value is inaccurate.

It's an edge case - here we're mainly interested in arriving at the style book via the style menu. The iframe will have loaded by the time a user navigates to the block styles I expect.